### PR TITLE
Add type to template components

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/utils/whatsapp/templates.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    """
+    Creates a parameter for each variable placeholder in the given content
+    """
+    params = []
+    seen = set()
+
+    for match in VARIABLE_RE.findall(content):
+        if match not in seen:
+            params.append({"type": _type})
+            seen.add(match)
```
Comment (rowanseymour) [inline temba/utils/whatsapp/templates.py:25]: if params can be repeated (e.g. `Hello {{1}} {{1}}`) like in some of our tests then you can just count variable placeholders

File: temba/utils/whatsapp/templates.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        comp_text = component.get("text", "")
+
+        if comp_type == "HEADER":
+            format = component.get("format", "TEXT")
+            params = []
+
+            if format == "TEXT":
+                params = _extract_params(comp_text)
+            elif format in ("IMAGE", "VIDEO", "DOCUMENT"):
+                params = [{"type": format.lower()}]
```
Comment (rowanseymour) [inline temba/utils/whatsapp/templates.py:49]: as I understand it none of these require parameters so are all safe to support

File: temba/utils/whatsapp/templates.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    # run through all our templates making sure they are present in our DB
+    seen = []
+    for template in wa_templates:
+        template_status = template["status"].upper()
+        if template_status not in STATUS_MAPPING:  # ignore if this is a status we don't know about
+            continue
+
+        components, all_supported = _extract_components(template["components"])
+
+        # TODO save components as a list... but for now organize them into a dict by type/index
```
Comment (rowanseymour) [inline temba/utils/whatsapp/templates.py:104]: the dict `"button.0": {}` way of organizing doesn't make so much sense any more... but keeping it as is for now

File: temba/utils/whatsapp/templates.py (hunk 2 above)
Comment (norkans7) [reply temba/utils/whatsapp/templates.py:49]: For now we mark them as not supported until we update the temba-components to support setting their values to send.

File: temba/utils/whatsapp/templates.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+def _extract_components(components) -> tuple:
+    """
+    Extracts components in our simplified format from payload of WhatsApp template components
+    """
+
+    extracted = []
+    all_supported = True
+
+    for component in components:
+        comp_type = component["type"]
```
Comment (norkans7) [inline temba/utils/whatsapp/templates.py:39]: We need to make this case insensitive either lower or upper that, currently I guess we upper that.

**Review by norkans7 — CHANGES_REQUESTED** [review]
make image , video and location attachment not supported for now

File: temba/utils/whatsapp/templates.py (hunk 4 above)
Comment (rowanseymour) [reply temba/utils/whatsapp/templates.py:39]: You've seen cases of this being lowercase?

Comment (norkans7) [reply temba/utils/whatsapp/templates.py]: I cannot remember that, but I think it is best to be case insensitive for that

**Pull request merged**